app/src/test/java/de/westnordost/streetcomplete/quests/barrier_type/AddDetectBarrierIntersectionTest.kt

```diff
@@ -0,0 +1,210 @@
+package de.westnordost.streetcomplete.quests.barrier_type
+
+import de.westnordost.streetcomplete.quests.TestMapDataWithGeometry
+import de.westnordost.streetcomplete.quests.crossing.AddCrossing
+import de.westnordost.streetcomplete.testutils.node
+import de.westnordost.streetcomplete.testutils.p
+import de.westnordost.streetcomplete.testutils.way
+import org.junit.Assert
+import org.junit.Test
+
+class AddDetectBarrierIntersectionTest {
```

Comment:
No point in testing the same algorithm twice. Instead of testing applicability of a certain quest type, you should just test the findNodesAtCrossingsOf algorithm separately. The only thing then left to test for `AddBarrierOnRoad` (etc.) are any specialities like the thing with the tunnel etc.

So in the end, this will be less test code and easier to maintain (i.e. most of `AddCrossingTest` can be deleted too)

Reply:
The benefit of such testing is that bugs in use of the algorithm will be found (even if algorithm itself would be fine) and bugs in tests are less likely due to testing on something more specific.

It would also catch at least some cases of bad filters and so on. 

I would keep it as is.

Reply:
alright